### ISSUE TITLE
[Feature Request] DarkLoadLibrary for DLL Imports

### ISSUE THREAD

**Issue by williamknows** [bats3c/DarkLoadLibrary#14]
Hey, great tool. Are there any near-term plans to add DarkLoadLibrary loading for a DLL's dependencies/imports? They're current just using LoadLibrary, and I can see you added a note to say support would (hopefully) be added in future (in `ldrutils.c`).

**Comment by hypervis0r** [comment]
do it urself

**Comment by physics-sec** [comment]
I tried it but there are still some cases where DarkLoadLibrary doesn't seem to work. I found some very weird errors with user32.dll if I remember correctly, the initialization routine (DLLMain) followed some very bizarre code paths for some reason

**Comment by bats3c** [comment]
Yes some Microsoft DLL's don't load correctly when using DarkLoadLibrary. I've not quite for to the bottom of this issue yet, I am currently investigating it.

**Comment by EddieIvan01** [comment]
> Yes some Microsoft DLL's don't load correctly when using DarkLoadLibrary. I've not quite for to the bottom of this issue yet, I am currently investigating it.

The reason is: you haven't handle the api set schema in your DIY GetProcAddress, here https://github.com/bats3c/DarkLoadLibrary/blob/master/DarkLoadLibrary/src/pebutils.c#L445

**Comment by bats3c** [comment]
This is not the reason for the issues, in the past `LdrGetProcedureAddress` from `ntdll` was in use and the issue still remained.
